packages/text-input/src/TextInput.spec.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { render, cleanup, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import TextInput, { State } from './TextInput';
+import { typeIs } from '@leafygreen-ui/lib';
+
+afterAll(cleanup);
+
+describe('packages/text-input', () => {
```

Comment:
Additional things to assert for:

- [ ] When you don't define an `optional` prop the optional text isn't present
- [ ] When optional is true and state is valid or error we don't see optional
- [ ] If state is Error and errorMessage is not set, we still see an error state
- [ ] If state is not Error and errorMessage is set, we don't see an error state

Reply:
thank for this- I felt like I was missing some important tests but I couldn't articulate what they were. I have added checks for all these things now!